src/elements/content-sidebar/SidebarNavButton.js

```diff
@@ -5,37 +5,40 @@
  */
 
 import * as React from 'react';
-import classNames from 'classnames';
-import PlainButton from '../../components/plain-button/PlainButton';
+import { NavLink, withRouter } from 'react-router-dom';
 import Tooltip from '../../components/tooltip/Tooltip';
 import './SidebarNavButton.scss';
 
 type Props = {
     children: React.Node,
+    history: Object,
     interactionTarget: string,
-    isSelected: boolean,
-    onClick: Function,
+    onToggle: Function,
+    sidebarView: string,
     tooltip: React.Node,
 };
 
-const SidebarNavButton = ({ tooltip, isSelected, onClick, interactionTarget, children }: Props) => {
-    const buttonClass = classNames('bcs-nav-btn', {
-        'bcs-nav-btn-is-selected': isSelected,
-    });
+const SidebarNavButton = ({ children, history, interactionTarget, onToggle, sidebarView, tooltip }: Props) => {
+    const linkPath = `/${sidebarView}`;
 
     return (
         <Tooltip position="middle-left" text={tooltip}>
-            <PlainButton
-                className={buttonClass}
+            <NavLink
+                activeClassName="bcs-nav-btn-is-selected"
+                className="bcs-nav-btn"
                 data-resin-target={interactionTarget}
                 data-testid={interactionTarget}
-                onClick={onClick}
-                type="button"
+                onClick={event => {
+                    if (history.location.pathname === linkPath) {
+                        onToggle(event);
```

Comment:
It might just be me but I get a little confused by the use of onToggle here. There's actually 2 kinds of toggles that can happen. 1 being toggling the sidebar open/closed, and 2 being toggling which tab is active. 

I know this isn't changed by you, but do you think it's worth switching to something like onSelect/Click?

Reply:
This area still needs a bit more work, regardless. I'll see what I can do to clean it up more.